tasks/libs/common/gitlab.py

```diff
@@ -231,22 +229,18 @@ def create_pipeline_schedule_variable(self, project_name, schedule_id, key, valu
         Create a variable for an existing pipeline schedule.
         """
         path = "/projects/{}/pipeline_schedules/{}/variables".format(quote(project_name, safe=""), schedule_id)
-        headers = {"Content-Type": "application/json"}
         data = {
             "key": key,
             "value": value,
         }
-        return self.make_request(path, headers=headers, data=json.dumps(data), json_output=True)
+        return self.make_request(path, data=data, json_output=True, json_input=True)
```

Comment:
We don't really need `json_input` here with your change, right?

Reply:
We do. The `make_request` function would otherwise make an api call with `data=` instead of `json=` which wouldn't work without headers. This is obviously non-ideal in terms of how the `make_request` function works, but I don't want to get into changing that right now, as it would require re-testing a lot of other tasks.

Reply:
Ah right, sorry, I just assumed this was using PUT too 😓